core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -125,22 +125,24 @@ object ConsumerPerformance {
     val startMs = System.currentTimeMillis
     var lastReportTime: Long = startMs
     var lastConsumedTime = System.currentTimeMillis
+    var currentTimeMillis = lastConsumedTime
     
-    while(messagesRead < count && System.currentTimeMillis() - lastConsumedTime <= timeout) {
+    while (messagesRead < count && currentTimeMillis - lastConsumedTime <= timeout) {
       val records = consumer.poll(100)
-      if(records.count() > 0)
+      if (records.count() > 0)
         lastConsumedTime = System.currentTimeMillis
-      for(record <- records) {
+      currentTimeMillis = System.currentTimeMillis
```

Comment:
Could we move this line to before line 132? That will save a System.currentTimeMillis call in 132.